### PR TITLE
Keep track of total, add window progress

### PR DESCRIPTION
**Pull request by jazzdan**
This doesn't use the new global _progress_ API yet, because my version of vscode doesn't seem to have it. Still trying to figure that out. Want to PR what I have in the meantime. This PR:

* Adds a global _status_ bar as a placeholder for the progress bar 
* Keeps track of the total in preparation for the progress bar

### CODE REVIEW

**Comment by jazzdan** [comment]
I'll add a gif in a second

**Comment by jazzdan** [comment]
Ugh, my gifshotter was disabled by Apple because it isn't 64 bit. It just shows a little text and a spinner on the bottom of the vscode window.

File: src/output_parser.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
+"use strict";
+
+import ParsedResult from "./parsed_result";
+
+export default class OutputParser {
+  public parse(line: string): ParsedResult | null {
+    if (this.isHeader(line)) {
```
Comment (jazzdan) [inline src/output_parser.ts]: There's probably a better way to do this?

Comment (nicks) [reply src/output_parser.ts]: seems fine for now tho

File: src/output_parser.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  }
+
+  private isHeader(line: string): boolean {
+    return line.startsWith("Pipeline ");
+  }
+
+  private parseHeader(line: string): ParsedResult {
+    const regex = /Pipeline [a-zA-Z]+\/[a-zA-Z]+:\d, Total: (\d+)/g;
+    let m = regex.exec(line);
+    console.log(m);
```
Comment (nicks) [inline src/output_parser.ts]: did you intend to leave this here?

File: src/output_parser.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    if (m) {
+      let total = m[1];
+
+      return {
+        total: parseInt(total)
```
Comment (nicks) [inline src/output_parser.ts]: usually better to do `Number(total)`.

in JS, `parseInt()` has super fuzzy semantics, and is very rarely preferable to `Number()`

File: src/output_parser.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+      total: 0
+    };
+  }
+
+  private isTotal(line: string): boolean {
+    return line.startsWith("TOTAL: ");
+  }
+
+  private parseTotal(line: string): ParsedResult {
+    let total = parseInt(line.split(":")[1].trim());
```
Comment (nicks) [inline src/output_parser.ts]: ditto

File: src/output_parser.ts (hunk 2 above)
Comment (jazzdan) [reply src/output_parser.ts]: No, thank you!

File: src/output_parser.ts (hunk 3 above)
Comment (jazzdan) [reply src/output_parser.ts]: yesss, the weird "base 10 is not the default" thing. Thanks for the catch.

**Pull request merged**